### DISCUSSION TITLE
multiple services get built, but just some of them got changes

### DISCUSSION THREAD

**Discussion by dragosbulugean** [aws/copilot-cli#3152] [Q&A]
We setup copilot for a couple LB services and AppRunner services.

Then we added the pipelines feature, which works great.

However, I noticed every service gets rebuilt on push, regardless of changes happening in the code of the specific service or not.

Do all of them get deployed regardless of changes? Is there a way to detect changes in the containers? Maybe a container hash based on its file contents or something?

Thank you for the clarification and Happy Holidays :)

**Comment by Lou1415926** [comment]
Hello @dragosbulugean ! Sorry for the late response, and a belated happy holidays to you too!

> Do all of them get deployed regardless of changes? Is there a way to detect changes in the containers? Maybe a container hash based on its file contents or something?

During the pipeline deployment stage, an update event is automatically triggered for all service stacks using the CloudFormation stack built from the pipeline build stage. Among all resources in the stack, your service resource may trigger an update, because the image tag that is used to reference the image changes (see [buildspec.yml](https://github.com/aws/copilot-cli/blob/d9ed362967525a3ceb82127b05a10d6470ac91cc/internal/pkg/template/templates/cicd/buildspec.yml#L57)). However, this shouldn't interrupt your service, as an update to service's task definition requires [no interruption](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-service.html#cfn-ecs-service-taskdefinition).

Does this address your concern?
